src/modules/timeline.js

```diff
@@ -163,36 +163,40 @@ export default function (playerInstance, options) {
-                    .addEventListener(eventOn, playerInstance.drawTimelinePreview.bind(playerInstance), false);
-                document.getElementById(playerInstance.videoPlayerId + '_fluid_controls_progress_container')
-                    .addEventListener(eventOff, function (event) {
-                        const progress = document.getElementById(playerInstance.videoPlayerId + '_fluid_controls_progress_container');
-                        if (typeof event.clientX !== 'undefined' && progress.contains(document.elementFromPoint(event.clientX, event.clientY))) {
-                            //False positive (Chrome bug when fast click causes leave event)
-                            return;
-                        }
-                        document.getElementById(playerInstance.videoPlayerId + '_fluid_timeline_preview_container').style.display = 'none';
-                        document.getElementById(playerInstance.videoPlayerId + '_fluid_timeline_preview_container_shadow').style.display = 'none';
-                    }, false);
-                playerInstance.generateTimelinePreviewTags();
+        const timelinePreview = playerInstance.displayOptions.layoutControls.timelinePreview;
+        if (timelinePreview &&
+            ((timelinePreview.type === 'VTT' && typeof timelinePreview.file === 'string') ||
+             (timelinePreview.type === 'static' && typeof timelinePreview.frames === 'object')) ) {
+            let eventOn = 'mousemove';
+            let eventOff = 'mouseleave';
+            if (playerInstance.mobileInfo.userOs) {
+                eventOn = 'touchmove';
+                eventOff = 'touchend';
+            }
+            document.getElementById(playerInstance.videoPlayerId + '_fluid_controls_progress_container')
+                .addEventListener(eventOn, playerInstance.drawTimelinePreview.bind(playerInstance), false);
+            document.getElementById(playerInstance.videoPlayerId + '_fluid_controls_progress_container')
+                .addEventListener(eventOff, function (event) {
+                    const progress = document.getElementById(playerInstance.videoPlayerId + '_fluid_controls_progress_container');
+                    if (typeof event.clientX !== 'undefined' && progress.contains(document.elementFromPoint(event.clientX, event.clientY))) {
+                        //False positive (Chrome bug when fast click causes leave event)
+                        return;
+                    }
+                    document.getElementById(playerInstance.videoPlayerId + '_fluid_timeline_preview_container').style.display = 'none';
+                    document.getElementById(playerInstance.videoPlayerId + '_fluid_timeline_preview_container_shadow').style.display = 'none';
+                }, false);
+            playerInstance.generateTimelinePreviewTags();
 
+            if (timelinePreview.type === 'VTT') {
```

Comment:
1. You can shift this `if` + `else` statement outside of the parent `if`
2. Prefer yoda style comparisons - `'VTT' === timelinePreview.type`